### PR TITLE
 CORE-37874 Added developer documentation.

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
Added developer documentation.  Note that the codebase doesn't yet follow all the conventions outlined. We've already agreed upon most of these things, but if you have any concern about anything that's been written, please comment. Otherwise, we should probably do the work to made the codebase consistent with the docs. If there's anything you think should be added, let me know.

## Related Issue
https://jira.corp.adobe.com/browse/CORE-37874

## Motivation and Context
Help developers get up to speed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+### Successful Execution
+
+If the command executes successfully, the promise is resolved with a result object. The content of the object depends on the command that was executed and, at times, the user's consent preferences. Note that if the user's consent is restrictive, the promise is still resolved, though possibly with sparse data, rather than the promise being rejected entirely. In other words, not being able to perform all tasks or provide all data due to limited user consent is not considered a failure. This is a very difficult but intentional design decision.
+
+Even if a command only has one piece of output, the output should always be provided as an attribute on the result object rather than the output being the result object itself. For example, the `createEventMergeId` command produces an event merge ID as follows:
+
+```js
+// Good: the event merge ID is provided as an attribute on result
+alloy("createEventMergeId", result => {
```
Comment (jfkhoury) [inline docs/api.md:62]: This should be:

```
alloy("createEventMergeId").then(result => console.log(result.eventMergeId));
```

Right?

Comment (Aaronius) [reply docs/api.md:62]: Yes. Thanks.

**Comment by Aaronius** [comment]
Yes, that would be good. I'll add those. Thanks.

**Comment by Aaronius** [comment]
I've since moved the developer content to the wiki: https://github.com/adobe/alloy/wiki

**Comment by jfkhoury** [comment]
> I've since moved the developer content to the wiki: https://github.com/adobe/alloy/wiki

Thank you @Aaronius, Love how it came out. Should we close this PR?

**Pull request merged**